### PR TITLE
[MNGSITE-548] Update subproject guide to new term

### PR DESCRIPTION
**Pull request by Bukama**
This PR updates the guide for multi-project setups to the new term "subproject".
It also consistently uses lowercase for the term subproject(s) in headings (except the page header).

### CODE REVIEW

**Review by elharo — CHANGES_REQUESTED** [review]
How do we redirect from the old URL to the new URL?

https://www.w3.org/Provider/Style/URI

**Comment by cstamas** [comment]
> How do we redirect from the old URL to the new URL?
> 
> https://www.w3.org/Provider/Style/URI

See https://github.com/apache/maven-site/blob/master/content/resources/xsd/.htaccess
We already do that for XSDs

**Comment by slawekjaranowski** [comment]
> > How do we redirect from the old URL to the new URL?
> > https://www.w3.org/Provider/Style/URI
> 
> Readded the old page with a link to the new one.

You should add redirect to: `content/filtered-resources/.htaccess` instead of page

File: content/resources/guides/mini/multi-subprojects.png
Comment (slawekjaranowski) [inline content/resources/guides/mini/multi-subprojects.png:1]: Can we try to use `puml` for it?

**Comment by Bukama** [comment]
> > > How do we redirect from the old URL to the new URL?
> > > https://www.w3.org/Provider/Style/URI
> > 
> > 
> > Readded the old page with a link to the new one.
> 
> You should add redirect to: `content/filtered-resources/.htaccess` instead of page

Thanks for the hint, updated (and removed the old one again :D )

File: content/resources/guides/mini/multi-subprojects.png
Comment (Bukama) [reply content/resources/guides/mini/multi-subprojects.png:1]: As my experience with puml is quite limited as of now, I would like to tackle this task while supporting you on #482.

But you are the boss :) I can try to do this after xmas.

Comment (slawekjaranowski) [reply content/resources/guides/mini/multi-subprojects.png:1]: picture is not an issues, can be improved later.

File: content/markdown/guides/mini/guide-multiple-subprojects-4.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+* Collects all the available subprojects to build
+* Sorts the subprojects into the correct build order
+* Selects which subprojects to build
+* Builds the selected subprojects in order
+
+### Collecting subprojects
+
+Subprojects collection starts from one aggregator project.
+That project defines the subprojects of which it consists using the `<subproject>` element.
+This is a recursive process, so aggregators can have child subproject which are aggregators themselves.
```
Comment (elharo) [inline content/markdown/guides/mini/guide-multiple-subprojects-4.md:42]: child subprojects

File: content/markdown/guides/mini/guide-multiple-subprojects-4.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+There are two ways to point at a starting POM.
+By default, Maven reads the `pom.xml` file in the working directory.
+Using the file parameter (`--file` or `-f`), you can point to another POM.
+
+### Sorting subprojects
+
+Because subprojects within a multi-project setup build can depend on each other, it is important that the reactor sorts
+all the
+projects in a way that guarantees any project is built before it is required.
```
Comment (elharo) [inline content/markdown/guides/mini/guide-multiple-subprojects-4.md:61]: any --> each

File: content/markdown/guides/mini/guide-multiple-subprojects-4.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+Subprojects inside a project can have two types of relationships: parent/child and dependency/dependent.
+
+When selecting a parent (aggregator), Maven automatically selects the child subprojects as well.
+Similarly, Maven excludes child subprojects of an excluded parent (aggregator).
+To prevent this recursive behaviour, combine `--projects` with `--non-recursive`.
+
+Maven knows about the dependencies between subprojects inside the multi-project setup.
+Using `--also-make`, Maven includes all dependencies of the selected projects in the build.
+Similarly, `--also-make-dependents` lets Maven include all subprojects which are dependent on the selected projects.
```
Comment (elharo) [inline content/markdown/guides/mini/guide-multiple-subprojects-4.md:111]: are dependent --> depend

File: content/markdown/guides/mini/guide-multiple-subprojects-4.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+#### Dealing with failures
+
+There are several ways to customize how the reactor deals with failures.
+`--fail-at-end` fails the build after building as many subprojects as possible.
+In this case, subprojects that do not depend on a failed subproject, will still be built.
```
Comment (elharo) [inline content/markdown/guides/mini/guide-multiple-subprojects-4.md:117]: subproject, will --> subproject will

File: content/markdown/guides/mini/guide-multiple-subprojects-4.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+This part of the Maven core does the following:
+
+* Collects all the available subprojects to build
+* Sorts the subprojects into the correct build order
+* Selects which subprojects to build
+* Builds the selected subprojects in order
+
+### Collecting subprojects
+
+Subprojects collection starts from one aggregator project.
```
Comment (elharo) [inline content/markdown/guides/mini/guide-multiple-subprojects-4.md:40]: Subproject collection

**Comment by Bukama** [comment]
Added @elharo suggestions and rebased (to resolve htaccess merge conflict)

**Review by elharo — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1130

**Pull request merged**